### PR TITLE
Fixes for mounting and matplotlib problems

### LINKED ISSUE #510: "XIO:  fatal IO error 25 (Inappropriate ioctl for device) on X server ":0"" in the testing stage of HelloWorld model
When running "python InnerEye/ML/runner.py --model=HelloWorld" on a local Ubuntu 20.04 machine without GPU, the error "XIO:  fatal IO error 25 (Inappropriate ioctl for device) on X server ":0"" occurred. This was caused by 'pool_outputs' on line 187 of ./InnerEye/ML/model_testing.py. I suspect this only occurs when there is only one testing data sample. The problem was resolved for me when I obtain 'pool_outputs' as:

pool_outputs = evaluate_model_predictions(process_id=0, 
                                       config=config, dataset=ds, results_folder=results_folder) 

[AB#4152](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/4152)

### LINKED ISSUE #513: Switch to using new AML mounting stack
Set environment variable "RSLEX_DIRECT_VOLUME_MOUNT": "true"

[AB#4158](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/4158)

### PR DESCRIPTION
**Pull request by ant0nsc**
- Fix for occasional issues in dataset mounting
- Fix for matplotlib problems on some machines
Closes #513 
Closes #510 

Please follow the guidelines for PRs contained [here](docs/pull_requests.md). Checklist:

- [x] Ensure that your PR is small, and implements one change.
- [x] Add unit tests for all functions that you introduced or modified.
- [x] Run PyCharm's code cleanup tools on your Python files.
- [x] Link the correct GitHub issue for tracking.
- [x] Update the [Changelog](CHANGELOG.md) file: Describe your change in terms of 
Added/Changed/Removed/... in the "Upcoming" section.
- [x] When merging your PR, replace the default merge message with a description of your PR,
and if needed a motivation why that change was required.

### CODE REVIEW

**Comment by ant0nsc** [comment]
This PR should not yet be merged, a first PR build failed in the dataset mounting.

File: Tests/AfterTraining/test_after_training.py [hunk 1]
```diff
@@ -368,7 +368,6 @@ def test_training_2nodes(test_output_dirs: OutputFolderForTests) -> None:
 
 
 @pytest.mark.after_training_2node
-@pytest.mark.skip("Test times out for unknown reasons.")
```
Comment (dumbledad) [inline Tests/AfterTraining/test_after_training.py:371]: Do we have an issue to reinvestigate this at a later date?

**Review by dumbledad — CHANGES_REQUESTED** [review]

**Review by Shruthi42 — APPROVED** [review]

**Review by JonathanTripp — APPROVED** [review]

File: Tests/AfterTraining/test_after_training.py (hunk 1 above)
Comment (dumbledad) [reply Tests/AfterTraining/test_after_training.py:371]: Ah, you are putting it back

**Review by dumbledad — APPROVED** [review]

**Review by JonathanTripp — APPROVED** [review]

**Review by dumbledad — APPROVED** [review]
Looks good

**Pull request merged**